tikv/raftstore/worker.go

```diff
@@ -270,6 +273,8 @@ func (r *splitCheckRunner) scanSplitKeys(spCheckerHost *splitCheckerHost, region
 			return nil, errors.Trace(err)
 		}
 	}
+	// Close the reader early, and in the `onKv` method above, some keys must be copied.
+	reader.Close()
```

Comment:
The reader doesn't hold any lock, we don't need to close it early.

Reply:
yes, thanks.